flame_fire_atlas/lib/flame_fire_atlas.dart

```diff
@@ -23,52 +21,87 @@ extension FireAtlasExtensions on Game {
   }
 }
 
-abstract class Selection {
+class Selection {
   String id;
   int x;
   int y;
   int w;
   int h;
 
+  Selection({
+    required this.id,
+    required this.x,
+    required this.y,
+    required this.w,
+    required this.h,
+  });
+
+  factory Selection.fromJson(Map<String, dynamic> json) {
+    return Selection(
+      id: json['id'],
+      x: json['x'],
+      y: json['y'],
+      w: json['w'],
+      h: json['h'],
+    );
+  }
+}
+
+abstract class BaseSelection {
+  final Selection info;
+
+  BaseSelection(this.info);
+
   Map<String, dynamic> toJson() {
     final Map<String, dynamic> json = {}
```

Comment:
`final json = <String, dynamic>{}`

also you could use the map literal and just return it?

```
return {
  id: info.id,
  ...
};
```

Reply:
also also you could consider adding json_build to the project :P

Reply:
Changed the first.

I am not a fan of code gen :(